atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -83,6 +84,17 @@
             Iterable<RangeRequest> rangeRequests);
 
     /**
+     * Same logic as {@link Transaction#getRanges(TableReference, Iterable)} but provides explicit control
+     * of parallelism. This should be favored over the former method to avoid unexpected concurrent calls
+     * hidden in the implementation.
+     */
+    @Idempotent
+    Iterable<BatchingVisitable<RowResult<byte[]>>> getRanges(
+            final TableReference tableRef,
+            Iterable<RangeRequest> rangeRequests,
+            RequestLimitedExecutorService executor);
```

Comment:
This technically can be just an `ExecutorService` but figured may be useful to force the user to use the limited version so they don't accidentally allow a request to grab all threads.